src/docs/sphinx/advancedExamples/validationStudies/mandel/Example.rst

```diff
@@ -0,0 +1,607 @@
+
+
+    def getGeometryFromXML(xmlFilePath):
+        tree = ElementTree.parse(xmlFilePath)
+
+        meshElement = tree.find('Mesh/InternalMesh')
+        dimensions = meshElement.get("xCoords")
+        dimensions = [float(i) for i in dimensions[1:-1].split(",")]
+        alen = dimensions[1]
+        dimensions = meshElement.get("zCoords")
+        dimensions = [float(i) for i in dimensions[1:-1].split(",")]
+        blen = dimensions[1]
+        dimensions = meshElement.get("nx")
+        dimensions = [float(i) for i in dimensions[1:-1].split(",")]
+        nx = dimensions[0]
+        dimensions = meshElement.get("nz")
+        dimensions = [float(i) for i in dimensions[1:-1].split(",")]
+        nz = dimensions[0]
+
+        return alen, blen, nx, nz
+
+
+    def main():
+        # File path
+        hdf5File1Path = "pressure_history.hdf5"
+        hdf5File2Path = "displacement_history.hdf5"
+        xmlFile1Path = "../../../../../../inputFiles/poromechanics/PoroElastic_Mandel_base.xml"
+        xmlFile2Path = "../../../../../../inputFiles/poromechanics/PoroElastic_Mandel_benchmark.xml"
+        
+        # Read HDF5
+        # Global Coordinate of Element Center
+        hf = h5py.File(hdf5File1Path, 'r')
+        xl = hf.get('pressure elementCenter')        
+        xcord = xl[0,:,0]
+        ycord = xl[0,:,1]
+        zcord = xl[0,:,2]        
+        pl = hf.get('pressure')       
+        tl = hf.get('pressure Time')  
+
+        # Global Coordinate of Nodal Point
+        hf = h5py.File(hdf5File2Path, 'r')
+        xl_node = hf.get('TotalDisplacement ReferencePosition')        
+        xcord_node = xl_node[0,:,0]
+        ycord_node = xl_node[0,:,1]
+        zcord_node = xl_node[0,:,2]
+        # Load Displacement Components
+        disp = hf.get('TotalDisplacement')       
+           
+        # Extract Mechanical Properties and Fracture Geometry from XML
+        hydromechanicalParameters = getHydromechanicalParametersFromXML(xmlFile1Path)
+        F = getAppliedTractionFromXML(xmlFile1Path)
+        La, Lb, na, nb = getGeometryFromXML(xmlFile2Path)
+        B = hydromechanicalParameters["skemptonCoefficient"]
+        nuu = hydromechanicalParameters["undrainedPoissonRatio"]
+        G = hydromechanicalParameters["shearModulus"]
+        p0 = 1./3./La*B*(1.+nuu)*F
+        u0 = -F*Lb*(1.0-nuu)/2./G/La
+        
+        # Extract Local Inform for Bottom Layer (z=0.01)
+        xlist = [] 
+        indlist = []    
+        for i in range(0,len(zcord)):
+            if abs(zcord[i]/(Lb/nb/2.)-1.) < 0.01 and abs(ycord[i]/0.05-1.) < 0.01:
+               xlist.append(xcord[i]/La)
+               indlist.append(i)            
+  
+        pplist = np.zeros([len(tl), len(xlist)]) 
+        for i in range(len(xlist)): 
+            ind = indlist[i]
+            pplist[:,i]= pl[:,ind]
+
+        zlist = [] 
+        indlist = [] 
+        for i in range(0,len(zcord_node)):
+            if xcord_node[i]==0.5 and ycord_node[i]==0.0:
```

Comment:
We should be a little cautious in using the current input files for Mandel's problem (`PoroElastic_Mandel_*`) as a validation example. We're missing the rigid constraint at the top/bottom surface, namely where the compressive loading is applied, in the problem definition. As such we cannot reproduce Mandel's analytical solution.

For example, the displacement in the `z`-direction (with reference to the sketch figure [here](https://github.com/GEOSX/GEOSX/blob/6dba08a3236d683fd5805c24411bae1e815429e5/src/docs/sphinx/advancedExamples/validationStudies/mandel/Example.rst)) should be a function of `z` and `time` only, with a zero gradient in the `x`-direction. Here, the script plotting the analytical solution extracts the numerical solution at nodes such that:
```python
if xcord_node[i]==0.5 and ycord_node[i]==0.0:
```
If `0.5` is changed to a different value the analytical solution will only be matched towards steady state -- the numerical solution being _behind_ (largest effect at `x = 0.0`) and _ahead_ (largest effect at `x = 1.0`) of the analytical solution for values lower and larger than `0.5` respectively , respectively. Similar comments apply to the displacement in the `x`-direction and `pressure`, both depending on `x` and `time` with a zero gradient in `z`. 

An easy fix would be to prescribe the analytical `z`-displacement at the loaded boundary instead of traction. This would require computing the analytical solution at `z = 1.0` for different time levels. The drawback of this approach is that if a user wants to modify hydromechanical parameters, such prescribed displacement condition needs to be changed accordingly. However, the python script that computes the analytical solution is already available.

If we all agree on the above strategy, we may also want to prescribe the exact initial solution corresponding to the undrained response. The pressure would be constant everywhere in the domain. The `x`-displacement would be linear in `x`, and the `z`-displacement would be linear in `z`.

Reply:
@castelletto1 Already apply these comments and please review this PR again.